seismostats/analysis/estimate_mc.py

```diff
@@ -318,3 +318,86 @@ def mc_max_curvature(
     )
     mc = bins[count.argmax()] + correction_factor
     return mc
+
+
+def mc_by_bvalue_stability(
+        sample: np.ndarray, delta_m: float, stability_factor: float = 0.1):
+    """
+    Estimates Mc using a test of stability.
+    The stability of the b-value is tested by default on half a magnitude unit
+    (in line with the 5x0.1 in the orginial paper). Users can change the range
+    for the stability test by changing the stability_factor (which gives the
+    size of the magnitude bin to use).
+
+    Source:
+        Cao, A., & Gao, S. S. (2002). Temporal variation of seismic b-values
+            beneath northeastern Japan island arc. Geophysical Research
+            Letters, 29(9), 1–3. https://doi.org/10.1029/2001gl013775
+
+    Args:
+        sample : np.array Vector of magnitudes
+        delta_m : float. Discretization of the magnitudes.
+        stability_factor : float. Magnitude bin to consider for the
+            stability test. Default is 0.1 to consider half a magnitude unit.
+
+    Returns:
+        mcs_test : np.ndarray. Tested completeness magnitudes
+        mc : float. Single best magnitude of completeness estimate
+        b : float. b-value associated with best_mc
+        n : int. Number of events greater than or equal to best_mc
+        sb_b_err : np.ndarray. Standard error of the b-value for all tested Mc
+        diff_b : np.ndarray. Difference between b-estimate and b-avg for each Mc
+        b_avgs : np.ndarray. Average b-value looking forward 5*stability_factor
+            for each step in Mc
+        bs : np.ndarray. Estimated b-value for each step in Mc
+        """
+    sample = bin_to_precision(sample, delta_x=delta_m)
```

Comment:
I think it would be better to test if it is binned correctly and then give a warning. Maybe this is for the future though, because this would require a test_binning function

Reply:
AFAIK this is inconsistent between methods. Some require the catalogue to be pre-binned-to-precision and some don't. I would keep this out for a different clean-up issue.